File: EconRentCarWeb/EconRentCar.DataModel/Cliente.cs
using EconRentCar.Common;
using EconRentCar.Core;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace EconRentCar.DataModel
{
    public class Cliente : IEntityBase
    {
        public Cliente()
        {
            Rentas = new HashSet<Renta>();
        }
        public Guid Id { get; set; }
        [Required]
        public string Nombres { get; set; }
        [Required]
        public string Apellidos { get; set; }
        [Required]
        public string CedulaCliente { get; set; }
        [Required]
        public string NoTArjetaCredito { get; set; }
        [DataType("money")]
        public double LimiteCredito { get; set; }
        [Required]
        public TipoPersona TipoPersona { get; set; }
        public bool Activo { get; set; }
        public virtual ICollection<Renta> Rentas { get; set; }
    }
}

File: EconRentCarWeb/EconRentCarClient/src/app/shared/models/cliente.ts
import { TipoPersona } from './enums';
import { Renta } from './renta';
import { BaseData } from '../dynamic-crud/models';

export class Cliente implements BaseData {
    public Id = '00000000-0000-0000-0000-000000000000';
    public Nombres = '';
    public Apellidos = '';
    public CedulaCliente = '';
    public NoTArjetaCredito = '';
    public LimiteCredito = 0.00;
    public TipoPersona = TipoPersona.Persona;
    public Activo = true;
    public Rentas: Renta[];
}
